src/connection.ts

```diff
@@ -116,15 +121,24 @@ export class ConnectionManager implements IConnectionManager {
       throw new Error('Unable to start the Language Server Process');
     }
 
-    this.languageServerProcess.stdout.on('data', data => {
-      this.logger.debug('OUTPUT: ' + data.toString());
+    switch (this.connectionConfiguration.protocol) {
+      case ProtocolType.TCP:
+        this.languageServerProcess.stdout.on('data', data => {
+          this.logger.debug('OUTPUT: ' + data.toString());
 
-      // If the language client isn't already running and it's sent the trigger text, start up a client
-      if (this.languageServerClient === undefined && data.toString().match('LANGUAGE SERVER RUNNING') !== null) {
-        this.languageServerClient = this.startLangClientTCP();
+          // If the language client isn't already running and it's sent the trigger text, start up a client
+          if (this.languageServerClient === undefined && data.toString().match('LANGUAGE SERVER RUNNING') !== null) {
+            this.languageServerClient = this.createLanguageClient();
+            this.extensionContext.subscriptions.push(this.languageServerClient.start());
+          }
+        });
+        break;
+      case ProtocolType.STDIO:
+        this.logger.debug('Starting STDIO client: ');
+        this.languageServerClient = this.createLanguageClient();
         this.extensionContext.subscriptions.push(this.languageServerClient.start());
```

Comment:
I tried to have this line not duplicated by moving it to line 142. However due to closures and how typescript/node/javacript/whatever works, line 142 is hit before anything in the case statement is evaluated and thus a null client is added before its actually initialized.